### PR TITLE
Remove duplicated code in arch/linux

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- Removes lots of duplicates code from arch/Linux
- Adds support for the `monitor` argument on linux

### CODE REVIEW

File: scapy/arch/linux.py [hunk 1]
```diff
@@ -408,102 +408,55 @@ def _flush_fd(fd):
...
+                         stderr=subprocess.STDOUT)
+    output, err = p.communicate()
+    match = re.search(br"mode:([a-zA-Z]*)", output.lower())
+    if match:
+        return plain_str(match.group(1))
+    return "unknown"
+
+
+def set_iface_monitor(iface, monitor):
+    mode = get_iface_mode(iface)
```
Comment (guedou) [inline scapy/arch/linux.py]: Could you add a docstring?

Comment (gpotter2) [reply scapy/arch/linux.py]: Done. I don't think those objects are tested

**Comment by guedou** [comment]
Can you check that the SuperSocket objects that you modified are correctly tested?

**Comment by gpotter2** [comment]
Ok the sockets are tested:
https://codecov.io/gh/secdev/scapy/compare/652b77bf12499451b47609b89abc663aa0f69c55...b2171e79834736dc361757636bd2aae2bfb71387/src/scapy/arch/linux.py?before=scapy/arch/linux.py#L464

However the monitor functions are not tested (neither are their on other platforms). But I can confirm they work on Linux with both pythons (tested them)

**Pull request merged**